OpenRA.Mods.Common/Traits/TunnelEntrance.cs

```diff
@@ -0,0 +1,72 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class TunnelEntranceInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[Desc("Offset to use as a staging point for actors entering or exiting the tunnel.",
+			"Should be at least Margin cells away from the actual entrance.")]
+		public readonly CVec RallyPoint = CVec.Zero;
+
+		[Desc("Cell radius to use as a staging area around the RallyPoint.")]
+		public readonly int Margin = 2;
+
+		[Desc("Offset to check for the corresponding map tunnel portal.")]
```

Comment:
Please use `"Offset to check for the corresponding TerrainTunnel portal cell(s)"`